workers/data_refinery_workers/processors/test_smasher.py

```diff
@@ -0,0 +1,348 @@
+        # Smoke test while we're here..
+        dataset.get_samples_by_experiment()
+        dataset.get_samples_by_species()
+        dataset.get_aggregated_samples()
+
+        # XXX: agg_type 'SPECIES' hangs on Linux, not OSX.
+        # Don't know why yet.
+        # for ag_type in ['ALL', 'EXPERIMENT', 'SPECIES']:
+        for ag_type in ['ALL', 'EXPERIMENT']:
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.aggregate_by = ag_type
+            dataset.save()
+
+            print ("Smashing " + ag_type)
+            final_context = smasher.smash(job.pk, upload=False)
+            # Make sure the file exists and is a valid size
+            self.assertNotEqual(os.path.getsize(final_context['output_file']), 0)
+            self.assertEqual(final_context['dataset'].is_processed, True)
+
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.is_processed = False
+            dataset.save()
+
+            # Cleanup
+            os.remove(final_context['output_file']) 
+
+        for scale_type in ['NONE', 'MINMAX', 'STANDARD', 'ROBUST']:
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.aggregate_by = 'EXPERIMENT'
+            dataset.scale_by = scale_type
+            dataset.save()
+
+            print ("Smashing " + scale_type)
+            final_context = smasher.smash(job.pk, upload=False)
+            # Make sure the file exists and is a valid size
+            self.assertNotEqual(os.path.getsize(final_context['output_file']), 0)
+            self.assertEqual(final_context['dataset'].is_processed, True)
+
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.is_processed = False
+            dataset.save()
+
+            # Cleanup
+            os.remove(final_context['output_file'])
+
+        # Stats
+        for scale_type in ['MINMAX', 'STANDARD']:
+            dataset = Dataset.objects.filter(id__in=relations.values('dataset_id')).first()
+            dataset.aggregate_by = 'EXPERIMENT'
+            dataset.scale_by = scale_type
+            dataset.save()
+
+            print("###")
+            print("# " + scale_type)
+            print('###')
+
+            final_context = smasher.smash(job.pk, upload=False)
+            final_frame = final_context['final_frame']
+            print("MEAN (per row):")
+            print(final_frame.mean(axis=1))
+            print("")
+
+            print("MIN (per row):")
+            print(final_frame.min(axis=1))
+            print("")
+
+            print("Max (per row):")
+            print(final_frame.max(axis=1))
+            print("")
+
+            print("STD (per row):")
+            print(final_frame.std(axis=1))
+            print("")
+
+            print("MEDIAN (per row):")
+            print(final_frame.median(axis=1))
+            print("")
```

Comment:
This seems like it'll be rather chatty and not provide much value unless someone actually looks into what is output and knows what would be expected. Perhaps a few asserts would be better here?